### PR TITLE
workflow: align imitation warm-start env contract

### LINKED ISSUE #1108: research: resolve #749 BC warm-start PPO artifact rescue
## Current Objective

This issue is now an **artifact-rescue issue** for the historical #749 BC warm-start PPO campaign.

The purpose is to determine whether the existing Slurm/Auxme trail produced recoverable durable evidence. It is **not** a fresh training issue.

Valid outcomes:

- `artifact_rescue_success`: durable dataset, BC checkpoint, PPO checkpoint, and comparison evidence can be recovered and promoted;
- `rerun_required`: the historical trail is not recoverable; open a clean rerun issue with current commit/config and close this one;
- `inconclusive_close`: the trail is stale and no durable evidence can be recovered; record why and close as inconclusive.

Do not launch new training from this issue unless a maintainer explicitly converts it into, or supersedes it with, a clean rerun issue.

## Current Resolution Mode

```yaml
slurm_state: artifact_rescue
issue: 1108
program: Hard-guarded hybrid learning
evidence_tier: smoke
resource: slurm
historical_branch: issue-1108-bc-warm-start-ppo
historical_job_id: 12462
known_result_root: output/slurm/issue1108-bcppo-job-12462/
known_log: output/slurm/12462-issue1108-bc-warm-start.out
required_decision:
  - artifact_rescue_success
  - rerun_required
  - inconclusive_close
durable_evidence_required:
  - expert trajectory dataset manifest or recoverability finding
  - BC checkpoint manifest or recoverability finding
  - PPO fine-tuned checkpoint manifest or recoverability finding
  - policy-analysis comparison against 27dbe5xu and b60iopxt, or recoverability finding
```

## Historical Context

The original research question was whether BC pretraining improves sample efficiency, final benchmark strength, both, or neither for a PPO v10 warm-start candidate.

The original launch packet and configs were:

- Parent: #749
- Launch packet: `docs/context/issue_749_bc_preinit_ppo_launch_packet.md`
- BC config: `configs/training/ppo_imitation/bc_pretrain_issue_749_v10_warm_start.yaml`
- PPO fine-tune config: `configs/training/ppo_imitation/ppo_finetune_issue_749_v10_warm_start.yaml`
- Historical expert comparator: `b60iopxt`
- Carry-forward v10 comparator: `27dbe5xu`

The historical execution trail was recorded as branch `issue-1108-bc-warm-start-ppo` and Slurm job `12462` on `a30/a30-gpu`, with local references to:

- result root: `output/slurm/issue1108-bcppo-job-12462/`
- log: `output/slurm/12462-issue1108-bc-warm-start.out`

These paths are historical/local references, not durable evidence by themselves.

## Rescue Scope

In scope now:

- Check whether the historical result root or durable artifact aliases still exist.
- Determine whether the expert trajectory dataset was collected and can be promoted or referenced durably.
- Determine whether the BC checkpoint exists and can be promoted or referenced durably.
- Determine whether the PPO fine-tuned checkpoint exists and can be promoted or referenced durably.
- Determine whether the canonical policy-analysis comparison against `27dbe5xu` and `b60iopxt` exists or can be reconstructed from durable artifacts.
- Record one of the three valid outcomes: `artifact_rescue_success`, `rerun_required`, or `inconclusive_close`.

Out of scope now:

- New reward design or architecture changes.
- Broad hyperparameter sweeps beyond the historical #749 candidate.
- Launching a fresh training job from this issue.
- Treating dry-run artifacts as benchmark evidence.
- Paper-facing claims from local-only files under `output/` without durable pointers.

### `artifact_rescue_success`

- [ ] Dataset `issue_749_b60iopxt_v10_eval_trajectories` exists with a validated durable manifest or equivalent durable pointer.
- [ ] BC checkpoint `issue_749_bc_preinit_v10_policy` exists with a durable manifest or equivalent durable pointer.
- [ ] PPO fine-tuned checkpoint `issue_749_ppo_finetune_v10_warm_start` exists with a durable manifest or equivalent durable pointer.
- [ ] Policy-analysis comparison against `27dbe5xu` and `b60iopxt` is recorded under a durable context/evidence note.
- [ ] Result states whether warm start improved final score, sample efficiency, both, or neither.
- [ ] Any local-only artifacts under `output/` are either promoted through a durable pointer or explicitly marked non-evidence.

### `rerun_required`

- [ ] Historical artifacts are not recoverable or are insufficient for durable evidence.
- [ ] The issue records the missing artifact classes and why they cannot be recovered.
- [ ] A new clean rerun issue is opened with current commit/config expectations, current artifact policy, and no dependence on stale local paths.
- [ ] This issue is closed as superseded by the rerun issue.

### `inconclusive_close`

- [ ] Historical artifacts are stale or unavailable.
- [ ] No durable dataset/checkpoint/comparison evidence can be recovered.
- [ ] The issue records why the result cannot support a positive, negative, or benchmark-strength conclusion.
- [ ] This issue is closed as inconclusive.

## Historical Commands

The following commands are retained only as historical context for the original experiment design. They are not current execution instructions unless copied into a new clean rerun issue.

Preflight path from the launch packet:

```bash
uv sync --group imitation
uv run python scripts/training/collect_expert_trajectories.py \
  --dataset-id issue_749_b60iopxt_v10_eval_trajectories \
  --policy-id ppo_expert_br06_v3_15m_all_maps_randomized_20260304T075200 \
  --episodes 1 \
  --scenario-config configs/scenarios/sets/ppo_full_maintained_eval_v1.yaml \
  --seeds 111 \
  --dry-run
uv run --group imitation python scripts/training/pretrain_from_expert.py \
  --config configs/training/ppo_imitation/bc_pretrain_issue_749_v10_warm_start.yaml \
  --dry-run
uv run python scripts/training/train_ppo_with_pretrained_policy.py \
  --config configs/training/ppo_imitation/ppo_finetune_issue_749_v10_warm_start.yaml \
  --dry-run
```

Original full execution shape:

```bash
uv run python scripts/training/collect_expert_trajectories.py \
  --dataset-id issue_749_b60iopxt_v10_eval_trajectories \
  --policy-id ppo_expert_br06_v3_15m_all_maps_randomized_20260304T075200 \
  --episodes 141 \
  --scenario-config configs/scenarios/sets/ppo_full_maintained_eval_v1.yaml \
  --seeds 111 112 113
uv run --group imitation python scripts/training/pretrain_from_expert.py \
  --config configs/training/ppo_imitation/bc_pretrain_issue_749_v10_warm_start.yaml
uv run python scripts/training/train_ppo_with_pretrained_policy.py \
  --config configs/training/ppo_imitation/ppo_finetune_issue_749_v10_warm_start.yaml
uv run python scripts/tools/policy_analysis_run.py \
  --policy ppo \
  --model-path output/benchmarks/expert_policies/issue_749_ppo_finetune_v10_warm_start_finetuned.zip \
  --seed-set eval \
  --max-seeds 3 \
  --output output/benchmarks/issue749_policy_analysis \
  --video-output output/recordings/issue749_policy_analysis \
  --all
```

## Added Value Estimation

- User value: medium if artifacts are rescued; low if the trail is unrecoverable but still worth closing cleanly.
- Benchmark value: high only if durable dataset/checkpoint/comparison evidence can be recovered.
- Maintenance value: high because this issue exercises the Slurm stale-trail closure protocol and prevents indefinite ambiguous execution issues.

## Effort Estimation

- Rough estimate: 2-6 hours for artifact-rescue classification.
- Best estimate: 3 hours.
- Main unknowns: availability of the historical result root, W&B/artifact permissions, checkpoint compatibility, and whether policy-analysis output exists durably.

## Complexity / Risk

- Artifact risk: high if local `output/` paths are missing or incomplete.
- Interpretation risk: high if partial artifacts are upgraded into benchmark evidence.
- Reproducibility risk: medium-high unless durable manifests/checksums can be recovered.
- Mitigation: classify the trail first; do not launch new training; close or split cleanly.

## Project Metadata

- Priority: Medium
- Effort (h): 3 for rescue classification
- Reviewed: 2026-05-29

## Portfolio Update

```yaml
issue_type: slurm-artifact-rescue
program: Hard-guarded hybrid learning
evidence_tier: smoke
resource: slurm
actionable_now: true
prerequisites: "access to historical Slurm/Auxme output root or durable artifact store"
```

## Artifact Promotion Management Update 2026-05-31

Tooling split: #1874 should provide the reusable local-artifact classification path that this rescue issue can use when checking historical #749 outputs.

For this issue, the planner should be used only for classification and manifest generation: present artifacts can be checksummed and reported, missing artifacts must be recorded as missing, and no upload or fresh training should happen from #1108 without a separate maintainer-approved rerun/promotion step.

The valid outcomes remain `artifact_rescue_success`, `rerun_required`, or `inconclusive_close`.

## Issue-Audit Split Update 2026-06-01

Mode: `split-parent-to-child`.

The reusable artifact planner from #1874 is available, and this machine does not currently expose the known historical #1108 output paths. The safe implementable child is #1961: classify artifact recoverability from known local paths and tracked durable references, then update this issue with `artifact_rescue_success`, `rerun_required`, or `inconclusive_close`. Do not launch fresh training from this parent.

### PR DESCRIPTION
**Pull request by ll7**
## Summary
Aligns the issue 749/issue 1108 imitation warm-start pipeline around a shared environment contract so trajectory collection, BC pretraining, and PPO fine-tuning use compatible scenario/env metadata after the latest `main` imitation-contract changes.

This is an unblocker PR related to issue 1108. It does **not** complete issue 1108 because the full SLURM experiment artifacts, durable pointers, and policy-analysis comparison are still outstanding.

## Linked Issues
- Relates to issue 1108
- Relates to issue 749

## What Changed
- Added the Auxme launcher for the issue-1108 BC warm-start PPO chain.
- Extended imitation configs with scenario/env contract metadata and CPU BC device control for the Auxme CUDA mismatch.
- Threaded training-config env overrides/factory kwargs through `scripts/training/imitation_env_contract.py` and the collection/BC/fine-tune scripts.
- Recorded the execution history and remaining evidence boundary in `docs/context/issue_1108_bc_warm_start_execution.md`.
- Added/updated targeted integration coverage for the BC device path and warm-start config loading.

## Why It Matters
- Added value: removes the observation-space/device mismatch that blocked the warm-start chain before the actual long-running PPO evidence can be trusted.
- Expected impact: makes the pipeline fail on real experiment outcomes rather than on avoidable collector/BC/fine-tune contract drift.
- Why this is worth merging now: it preserves a validated, latest-main-compatible unblocker while issue 1108 remains a SLURM/artifact follow-up.

## Validation / Proof
- Commands run:
  - `uv run python -m py_compile robot_sf/training/imitation_config.py scripts/training/imitation_env_contract.py scripts/training/collect_expert_trajectories.py scripts/training/pretrain_from_expert.py scripts/training/train_ppo_with_pretrained_policy.py`
  - `bash -n SLURM/Auxme/issue_1108_bc_warm_start.sl`
  - `uv run ruff check robot_sf/training/imitation_config.py scripts/training/imitation_env_contract.py scripts/training/collect_expert_trajectories.py scripts/training/pretrain_from_expert.py scripts/training/train_ppo_with_pretrained_policy.py tests/training/test_imitation_env_contract.py tests/training/test_pretrain_from_expert.py tests/training/test_train_ppo_with_pretrained_policy.py tests/integration/test_ppo_pretraining_pipeline.py`
  - `uv run ruff format --check robot_sf/training/imitation_config.py scripts/training/imitation_env_contract.py scripts/training/collect_expert_trajectories.py scripts/training/pretrain_from_expert.py scripts/training/train_ppo_with_pretrained_policy.py tests/training/test_imitation_env_contract.py tests/training/test_pretrain_from_expert.py tests/training/test_train_ppo_with_pretrained_policy.py tests/integration/test_ppo_pretraining_pipeline.py`
  - `uv run pytest tests/training/test_imitation_env_contract.py tests/training/test_pretrain_from_expert.py tests/training/test_train_ppo_with_pretrained_policy.py tests/integration/test_ppo_pretraining_pipeline.py -q`
  - `BASE_REF=origin/main scripts/dev/check_docs_proof_consistency_diff.sh`
  - `PYTEST_NUM_WORKERS=8 BASE_REF=origin/main DISPLAY= MPLBACKEND=Agg SDL_VIDEODRIVER=dummy scripts/dev/pr_ready_check.sh`
- Evidence that the change works here:
  - targeted pytest: `22 passed`
  - final readiness gate: `3627 passed, 10 skipped, 3 warnings in 429.49s`
  - doc/proof consistency: passed for 12 changed files
  - touched-definition docstring TODO check: clean
- Benchmarks or smoke tests, if applicable:
  - No benchmark-strength SLURM rerun from this host. `local.machine.md` has `allow_slurm_submission: false`.

## Risks / Rollout
- Compatibility risks: config fields are additive and default to current behavior when omitted.
- Failure modes: the actual experiment can still fail on model quality, long-run training stability, or artifact publication after this unblocker.
- Rollback or fallback plan: revert this PR and keep the experiment blocked on the prior collector/BC/PPO contract mismatch.

## Docs / Provenance
- Updated docs:
  - `docs/context/README.md`
  - `docs/context/issue_1108_bc_warm_start_execution.md`
  - `docs/context/issue_749_bc_preinit_ppo_launch_packet.md`
- Relevant design or provenance notes: `docs/context/issue_1108_bc_warm_start_execution.md` records the earlier SLURM attempts and explicitly separates local cache output from durable evidence.
- Any assumptions that need to be preserved: worktree-local `output/` coverage files are disposable; issue completion still requires durable dataset/checkpoint pointers plus policy-analysis comparison against `27dbe5xu` and `b60iopxt`.

## Follow-Up Issues
- Deferred work: complete the experiment on a machine allowed to submit/inspect Auxme SLURM jobs, then publish durable artifacts and analysis results.
- Issues opened for follow-up: none; issue 1108 remains open for the deferred experiment evidence.

## Reviewer Notes
- Please verify closely that the collector, BC pretrain, and PPO fine-tune scripts all resolve the same scenario/env contract.
- Known limitation: this PR is intentionally draft and does not claim benchmark-strength experiment completion.

### CODE REVIEW

**Comment by ll7** [comment]
Closing and reopening from a neutral branch name because GitHub kept treating the issue branch as a closing reference for issue 1108, which is incorrect for this unblocker PR.
